src/components/WishlistEditor/WishlistEditor.js

```diff
@@ -0,0 +1,155 @@
+import React, { useState } from 'react';
+import { Link, Redirect } from 'react-router-dom';
+import {
+  Paper,
+  TextField,
+  Button,
+  FormControl,
+  IconButton,
+} from '@material-ui/core/';
+import SaveIcon from '@material-ui/icons/Save';
+import { makeStyles } from '@material-ui/core/styles';
+import CloseIcon from '@material-ui/icons/Close';
+
+import './WishlistEditor.css';
+
+const useStyles = makeStyles({
+  wishlistEditorContainer: {
+    height: '100%',
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'center',
+  },
+  paper: {
+    width: '50%',
+    padding: '3%',
+    display: 'flex',
+    justifyContent: 'space-between',
+  },
+  input: {
+    marginBottom: '16px',
+  },
+});
+
+const WishlistEditor = ({ handleSendWishlist }) => {
+  const classes = useStyles();
+
+  const [name, setName] = useState('');
+  const [author, setAuthor] = useState('');
+  const [items, setItems] = useState(['']);
+  const [nameErrorText, setNameErrorText] = useState('');
+  const [authorErrorText, setAuthorErrorText] = useState('');
+  const [itemErrorText, setItemErrorText] = useState('');
+  const [redirect, setRedirect] = useState('');
+
+  const handleNameChange = (event) => {
+    setNameErrorText('');
+    setName(event.target.value);
+  };
+
+  const handleAuthorChange = (event) => {
+    setAuthorErrorText('');
+    setAuthor(event.target.value);
+  };
+
+  const handleItemsChange = (index, event) => {
+    setItemErrorText('');
+    items[index] = event.target.value;
+    if (items[items.length - 1].trim()) items.push('');
+    setItems([...items]);
+  };
+
+  const handleSubmit = (event) => {
+    const filteredItems = items
+      .map((item) => item.trim())
+      .filter((item) => item);
```

Comment:
Array.filter() is used to remove items that are set to a blank string from the array.  That's fine, but since we're storing blank strings in the items array, we have to do this procedure of using `.filter((item) => item)` several more times throughout this document to get the desired result.  One of the principles of DRY (Don't Repeat Yourself) programming is that if you have to write something out three times, it's worth encapsulating somehow - so, in the long run, it would probably be good to either: a) encapsulate that functionality in a reusable way, i.e. with a function: 
`let cleanArr = (arr)=> arr.filter(item=>item);`
or b) decide on some other way of working with the items state value so that it is an array that doesn't contain blank strings at any point and thus avoids the issue that way.